text_collector_examples/storcli.py

```diff
@@ -48,27 +48,28 @@ def main(args):
 
         for controller in data:
             response = controller['Response Data']
-            
-            handle_common_controller(response)
+            (controller_index, baselabel) = get_basic_controller_info(response)
+
+            handle_common_controller(response,controller_index, baselabel)
             if response['Version']['Driver Name'] == 'megaraid_sas':
-                handle_megaraid_controller(response)
+                handle_megaraid_controller(response,controller_index, baselabel)
             elif response['Version']['Driver Name'] == 'mpt3sas':
-                handle_sas_controller(response)
+                handle_sas_controller(response,controller_index, baselabel)
     except KeyError:
         pass
 
     print_all_metrics(metric_list)
 
-def handle_common_controller(response):
-    (controller_index, baselabel) = get_basic_controller_info(response)
-
-    # Split up string to not trigger CodeSpell issues
-    if 'ROC temperature(Degree Celc' + 'ius)' in response['HwCfg'].keys():
-        response['HwCfg']['ROC temperature(Degree Celsius)'] = response['HwCfg'].pop('ROC temperature(Degree Celc' + 'ius)')
-    add_metric('temperature', baselabel, int(response['HwCfg']['ROC temperature(Degree Celsius)']))
+def handle_common_controller(response,controller_index, baselabel):
+    try:
+        # Split up string to not trigger CodeSpell issues
+        if 'ROC temperature(Degree Celc' + 'ius)' in response['HwCfg'].keys():
+            response['HwCfg']['ROC temperature(Degree Celsius)'] = response['HwCfg'].pop('ROC temperature(Degree Celc' + 'ius)')
+        add_metric('temperature', baselabel, int(response['HwCfg']['ROC temperature(Degree Celsius)']))
```

Comment:
From what I can tell, the only possible exception here would be a ValueError coming from the `int` function.  Since `add_metric` already handles a string to float conversion and the associated possible error, maybe it would be better to just remove the call to `int()` and just pass the string value from the dict directly to `add_metric`?  Then you should get the same error handling but without needed the try block.  WDYT?